apps/automation-v2/src/main.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  TaskContext,
+  TaskException,
+  createTaskRunner,
+  fallibleWebhookTask,
+  taskSuccess,
+} from '@mountainpath9/overlord';
+
+import { batchLiquidate } from '@/tlc/batch-liquidate';
+
+import { CONFIG as CONFIG_TESTNETS } from '@/config/testnets';
+import { CONFIG as CONFIG_PRODNETS } from '@/config/prodnets';
+
+import { DISCORD_WEBHOOK_URL_KEY, connectDiscord } from '@/common/discord';
+import { MAINNET, SEPOLIA } from '@/chains';
+import { checkLowEthBalance } from '@/common/eth-auto-checker';
+
+async function main() {
+  const runner = createTaskRunner();
+
+  const mode = await runner.config.requireString('temple_networks');
+  const isProdnet = mode === 'prodnets';
+  const config = isProdnet ? CONFIG_PRODNETS : CONFIG_TESTNETS;
+
+  runner.setTaskExceptionHandler(discordNotifyTaskException);
+
+  runner.addPeriodicTask({
+    id: 'tlc-batch-liquidate',
+    cronSchedule: '*/10 * * * *',
+    action: async (ctx) => {
+      const res = await batchLiquidate(ctx, config.tlcBatchLiquidate);
+      await checkLowEthBalance(
```

Comment:
The checkLowEthBalance code structure started feeling a bit unwieldy to me. In swapbot, I just check for a low eth balance at the end of a successful action:

https://github.com/TempleDAO/swapbot/blob/c66e4dc71483207f2f4020681c0db7761f4b7e76/automation/src/swapbot.ts#L72

Perhaps consider this.